src/superqt/utils/_errormsg_context.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+import traceback
+from contextlib import AbstractContextManager
+from typing import TYPE_CHECKING, cast
+
+from qtpy.QtWidgets import QErrorMessage, QMessageBox, QWidget
+
+if TYPE_CHECKING:
+    from types import TracebackType
+
+
+class exceptions_as_dialog(AbstractContextManager):
+    """Context manager that shows a dialog when an exception is raised.
+
+    To determine whether an exception was raised or not, check the `exception`
+    attribute after the context manager has exited.  If `use_error_message` is `False`
+    (the default), you can also access the `widget` attribute to get/manipulate the
+    `QMessageBox` instance.
+
+    Parameters
+    ----------
+    exceptions : type[BaseException] | tuple[type[BaseException], ...], optional
+        The exception(s) to catch, by default `Exception` (i.e. all exceptions).
+    icon : QMessageBox.Icon, optional
+        The icon to show in the QMessageBox, by default `QMessageBox.Icon.Critical`
+    title : str, optional
+        The title of the `QMessageBox`, by default "An error occurred"
+    msg_template : str, optional
+        The message to show in the `QMessageBox`. The message will be formatted
+        using three variables:
+        - `exc_value`: the exception instance
+        - `exc_type`: the exception type
+        - `tb`: the traceback as a string
+        The default template is the content of the exception: `"{exc_value}"`
+    buttons : QMessageBox.StandardButton, optional
+        The buttons to show in the `QMessageBox`, by default
+        `QMessageBox.StandardButton.Ok`
+    parent : QWidget | None, optional
+        The parent widget of the `QMessageBox`, by default `None`
+    use_error_message : bool | QErrorMessage, optional
+        Whether to use a `QErrorMessage` instead of a `QMessageBox`. By default
+        `False`. `QErrorMessage` shows a checkbox that the user can check to
+        prevent seeing the message again (based on the text of the formatted
+        `msg_template`.) If `True`, the global `QMessageError.qtHandler()`
+        instance is used to maintain a history of dismissed messages. You may also pass
+        a `QErrorMessage` instance to use a specific instance. If `use_error_message` is
+        True, `parent` is ignored.
```

Comment:
"Is not false?" As parent, it is also ignored when provided QErrorMessage? 

How it integrates with the change in global style (like in napari)?

Reply:
> As parent, it is also ignored when provided QErrorMessage?

Yes, ignored there as well.  If someone is handling the construction of the QErrorMessage instance, I think they should also handle passing the parent to the constructor (and we shouldn't override it).

> How it integrates with the change in global style (like in napari)?

 at least on macos, QErrorMessage actually uses a native dialog, anyway, so it can't be styled:

```
from qtpy.QtWidgets import QApplication, QErrorMessage

app = QApplication([])
err = QErrorMessage()
err.setStyleSheet("QErrorMessage { background-color: red }")
err.showMessage("This is an error message")
app.exec_()

```

<img width="372" alt="Screenshot 2023-08-21 at 11 28 25 AM" src="https://github.com/pyapp-kit/superqt/assets/1609449/131b3eb5-b6fb-43d5-9d7b-715cd62d7871">

Reply:
Ok. Stil docstring need update (from my point of understanding)

Reply:
changed to Is not False. 👍